index.js

```diff
@@ -0,0 +1,252 @@
+      final['recommended'] = tempReccomended;
+
+    })
+    .then((empty, err) => { //then we look into characteristics
+
+      var charComplete = {}; //container for characteristics
+      if (err) {
+        throw err;
+      } else {
+        Characteristics_data.findAll({ //finds all char associated with the product id
+          where: {
+            product_id: productId
+          }
+        })
+          .then((data, err) => {
+
+            if (err) {
+              throw err;
+            } else {
+              var list = data.map((entry) => { //finds all reviews asoicated using the table
+                return Characteristics_reviews.findAll({
+                  where: {
+                    characteristic_id: entry.id
+                  }
+                })
+              });
+              Promise.all(list)
+                .then((arrOfValues, err) => {
+
+                  if (err) {
+                    throw err;
+                  } else {
+                    for (var i = 0; i < arrOfValues.length; ++i) { //calculates avg
+                      var avg = 0;
+                      for (var n = 0; n < arrOfValues[i].length; ++n) {
+                        avg += arrOfValues[i][n].value;
+                      }
+                      avg = avg/arrOfValues[i].length;
+                      charComplete[data[i].name] = {
+                        id: data.id,
+                        average: avg
+                      }
+                    }
+                    final['characteristics'] = charComplete;
+                    res.send(final);
+                  }
+                })
+
+            }
+          });
+
+      }
+
+    })
+    .catch((err) => {
+      res.send(err);
+    })
+});
+
+app.post('/reviews', (req, res) => {
+  Info.create({ //creates review in db
+    product_id: req.body.product_id,
+    rating: req.body.rating,
+    summary: req.body.summary,
+    body: req.body.body,
+    date: Date.now(),
+    recommended: req.body.recommend,
+    reviewer_name: req.body.name,
+    reviewer_email: req.body.email
+  })
+  .then((data, err) => {
+
+    var listA = req.body.photos.map((entry) => { //creates photos in db associated with that review
+      return Info_photos.create({
+        review_id: data.id,
+        url: entry
+      })
+    });
+    var photos = Promise.all(listA);
+
+    var listB = Object.keys(req.body.characteristics).map((entry) => { //creates connection between review and char
+      return Characteristics_reviews.create({
+        characteristic_id: entry,
+        review_id: data.id,
+        value: req.body.characteristics[entry]
+      })
+    });
+    var reviews_char = Promise.all(listB);
+
+    Promise.all(listA, listB)
+      .then((entry, err) => {
+
+        if (err) {
+          throw err;
+        } else {
+          res.send(entry);
+        }
+
+      });
+
+  })
+  .catch((err) => {
+    res.send(err);
+  });
+});
+
+app.put('/reviews/:review_id/helpful', (req, res) => {
```

Comment:
For this request, is it expecting the the review_id within the body?
Or is it expecting to the review_id from the URL string as a request parameter?

It seems that you are retrieving the review_id from the body on line 219.